R/RLearner_classif_mxff.R

```diff
@@ -0,0 +1,416 @@
+    If `eval.data` is specified, `validation.set` will be ignored. Note that `eval.data` is passed
+    to `mx.model.FeedForward.create` unchanged to provide unconstrained usability of the underlying
+    learner. In particular, this implies that `array.layout` is not adapted when using convolution,
+    so `eval.data` needs to be provided in the right format.
+    If `validation.set` is specified, it is sampled randomly using `R`'s `sample`.
+    If `early.stop.badsteps` is specified and `epoch.end.callback` is not specified,
+    early stopping will be used using `mx.callback.early.stop` as `epoch.end.callback` with the
+    learner's `eval.metric`. In this case, `early.stop.badsteps` gives the number of `bad.steps` in
+    `mx.callback.early.stop` and `early.stop.maximize` gives the `maximize` parameter in
+    `mx.callback.early.stop`. Please note that when using `early.stop.badsteps`, `eval.metric` and
+    either `eval.data` or `validation.set` should be specified.
+    "
+  )
+}
+
+#' @export
+trainLearner.classif.mxff = function(.learner, .task, .subset, .weights = NULL,
+  layers = 1L, num.layer1 = 1L, num.layer2 = 1L, num.layer3 = 1L,
+  act1 = "tanh", act2 = "tanh", act3 = "tanh", act.out = "softmax",
+  conv.data.shape = NULL, conv.layer1 = FALSE, conv.layer2 = FALSE, conv.layer3 = FALSE,
+  conv.kernel11 = NULL, conv.kernel21 = NULL, conv.kernel31 = NULL,
+  conv.kernel12 = NULL, conv.kernel22 = NULL, conv.kernel32 = NULL,
+  conv.stride11 = NULL, conv.stride21 = NULL, conv.stride31 = NULL,
+  conv.stride12 = NULL, conv.stride22 = NULL, conv.stride32 = NULL,
+  conv.dilate11 = NULL, conv.dilate21 = NULL, conv.dilate31 = NULL,
+  conv.dilate12 = NULL, conv.dilate22 = NULL, conv.dilate32 = NULL,
+  conv.pad11 = NULL, conv.pad21 = NULL, conv.pad31 = NULL,
+  conv.pad12 = NULL, conv.pad22 = NULL, conv.pad32 = NULL,
+  pool.kernel11 = NULL, pool.kernel21 = NULL, pool.kernel31 = NULL,
+  pool.kernel12 = NULL, pool.kernel22 = NULL, pool.kernel32 = NULL,
+  pool.stride11 = NULL, pool.stride21 = NULL, pool.stride31 = NULL,
+  pool.stride12 = NULL, pool.stride22 = NULL, pool.stride32 = NULL,
+  pool.pad11 = NULL, pool.pad21 = NULL, pool.pad31 = NULL,
+  pool.pad12 = NULL, pool.pad22 = NULL, pool.pad32 = NULL,
+  pool.type1 = "max", pool.type2 = "max", pool.type3 = "max",
+  dropout = NULL, symbol = NULL, validation.set = NULL, eval.data = NULL, early.stop.badsteps = NULL,
+  epoch.end.callback = NULL, early.stop.maximize = TRUE, array.layout = "rowmajor", ...) {
+  # transform data in correct format
+  d = getTaskData(.task, subset = .subset, target.extra = TRUE)
+  y = as.numeric(d$target) - 1
+  x = data.matrix(d$data)
+
+  # construct validation data
+  if (is.null(eval.data) & !is.null(validation.set)) {
+    eval.data = list()
+    n = dim(x)[1]
+    val.ind = sample(n, floor(n * validation.set))
+    eval.data$label = y[val.ind]
+    y = y[-val.ind]
+    eval.data$data = x[val.ind,]
+    x = x[-val.ind,]
+  }
```

Comment:
You might want to use `makeResampleDesc("Holdout", ...)` to make use of stratification